ai_genomics/pipeline/patent_data/get_ai_genomics_patents.py

```diff
@@ -0,0 +1,152 @@
+"""Script to query google patent data with USPTO
+AI patent ids and identify patent ids that have
+genomics-related classification codes.
+
+To run script,
+export GOOGLE_APPLICATION_CREDENTIALS="/path/to/credentials.json"
+python get_ai_genomics_patents.py
+"""
+#################################################
+from ai_genomics import bucket_name, config
+from ai_genomics.utils.patent_data.get_ai_genomics_patents_utils import est_conn
+from ai_genomics.getters.data_getters import (
+    s3,
+    load_s3_data,
+    save_to_s3,
+    get_s3_dir_files,
+)
+
+import pandas as pd
+import numpy as np
+
+from google.api_core.exceptions import Forbidden
+import uuid
+
+#################################################
+
+
+def make_query_chunks(uspto_patent_ids, n_chunks: int, table: str) -> list:
+    """Generate BigQuery query chunks based on USPTO AI patent IDS.
+
+    Args:
+        uspto_patent_ids: List of patent IDs.
+        n_chunks (int): The number of chunks generated from uspto_patent_ids.
+                        Chunk string size must be less than 1024.00K characters,
+                        including comments and white space characters. This means
+                        there needs to be approx. 85 query chunks to accomodate
+                        string size limits.
+
+    Returns:
+        uspto_queries (list): List of n_chunk size of BigQuery queries.
+    """
+    uspto_queries = []
+    uspto_patent_chunks = np.array_split(
+        uspto_patent_ids, n_chunks
+    )  # split based on query string limits
+
+    for uspto_patent_chunk in uspto_patent_chunks:
+        ids = "'" + "', '".join([str(i) for i in uspto_patent_chunk]) + "'"
+        q = (
+            f"SELECT application_number "
+            f"FROM `{table}` "
+            f"WHERE REGEXP_EXTRACT(application_number, r'[0-9]+') IN ({ids})"
+        )
+
+        uspto_queries.append(q)
+
+    if [
+        len(uspto_query_chunk)
+        for uspto_query_chunk in uspto_queries
+        if len(uspto_patent_chunk) > 1024000
+    ] != []:
+        print("A query chunk(s) too large - make n_chunk int larger.")
```

Comment:
A better way would be to handle this on the fly with dynamic chunking (making the query length the max based on the chunk size and then check length and truncate if needed).

Reply:
Actually it would be simpler to add IDs until the `len(ids) + len(query)` is almost the query limit then start a new chunk. This would eliminate the need for the user to define a chunk size up front.